src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeed.cs

```diff
@@ -265,6 +272,273 @@ public TextSyndicationContent Title
+                    _skipHours = skipHours;
+                }
+
+                return _skipHours;
+            }
+        }
+
+        internal Collection<string> InternalSkipDays => _skipDays;
+
+        public Collection<string> SkipDays
+        {
+            get
+            {
+                if (_skipDays == null)
+                {
+                    var skipDays = new Collection<string>();
+                    TryReadSkipDaysFromExtension(ElementExtensions, skipDays);
+                    _skipDays = skipDays;
+                }
+
+                return _skipDays;
+            }
+        }
+
+        internal SyndicationTextInput InternalTextInput => _textInput;
+
+        public SyndicationTextInput TextInput
+        {
+            get
+            {
+                if (_textInput == null)
+                {
+                    _textInput = TryReadTextInputFromExtension(ElementExtensions);
+                }
+
+                return _textInput;
+            }
+            set
+            {
+                _textInput = value;
+            }
+        }
+
+        private SyndicationLink TryReadDocumentationFromExtension(SyndicationElementExtensionCollection elementExtensions)
+        {
+            SyndicationElementExtension documentationElement = elementExtensions
+                                      .Where(e => e.OuterName == Rss20Constants.DocumentationTag && e.OuterNamespace == Rss20Constants.Rss20Namespace)
+                                      .FirstOrDefault();
+
+            if (documentationElement == null)
+                return null;
+
+            using (XmlReader reader = documentationElement.GetReader())
+            {
+                SyndicationLink documentation = Rss20FeedFormatter.ReadAlternateLink(reader, BaseUri, SyndicationFeedFormatter.DefaultUriParser, preserveAttributeExtensions: true);
+                return documentation;
+            }
+        }
+
+        private TimeSpan? TryReadTimeToLiveFromExtension(SyndicationElementExtensionCollection elementExtensions)
+        {
+            SyndicationElementExtension timeToLiveElement = elementExtensions
+                                      .FirstOrDefault(e => e.OuterName == Rss20Constants.TimeToLiveTag && e.OuterNamespace == Rss20Constants.Rss20Namespace);
+
+            if (timeToLiveElement == null)
+                return null;
+
+            using (XmlReader reader = timeToLiveElement.GetReader())
+            {
+                string value = reader.ReadElementString();
+                if (int.TryParse(value, out int timeToLive))
+                {
+                    if (timeToLive >= 0)
+                    {
+                        return TimeSpan.FromMinutes(timeToLive);
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                }
+                else
+                {
+                    return null;
```

Comment:
Nit: This could be simplified, e.g.:

```c#
using (XmlReader reader = timeToLiveElement.GetReader())
{
    string value = reader.ReadElementString();
    if (int.TryParse(value, out int timeToLive) && timeToLive >= 0)
    {
        return TimeSpan.FromMinutes(timeToLive);
    }

    return null;
}
```